### ISSUE TITLE
usage with mixcr files

### ISSUE THREAD

**Issue by guillemsanchezsanchez1996** [phbradley/tcr-dist#41]
Dear Phil,
After reading your paper I would be interested in checking identifying shared motifs in a particular subset of sorted T cells that undergo strong TCR signaling during their development in thymus. I'm quite new with python and I have been able to set up the enviroment and installl packages and all dependecies in windows using WSL.
Unfortunately I have only single chain files (alpha and beta) from several subjects. Is there a way I can adapt the input to perform the analysis only with single chain files?
Thanks in advance for your help :)

Guillem Sanchez

**Comment by phbradley** [comment]
Hi Guillem,
Can you provide an example of the format of your input files? Do you have gene expression data or is this just TCR sequences?
Take care,
Phil
________________________________
From: guillemsanchezsanchez1996 ***@***.***>
Sent: Monday, February 7, 2022 5:05 AM
To: phbradley/tcr-dist ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [phbradley/tcr-dist] usage with mixcr files (Issue #41)


Dear Phil,
After reading your paper I would be interested in checking identifying shared motifs in a particular subset of sorted T cells that undergo strong TCR signaling during their development in thymus. I'm quite new with python and I have been able to set up the enviroment and installl packages and all dependecies in windows using WSL.
Unfortunately I have only single chain files (alpha and beta) from several subjects. Is there a way I can adapt the input to perform the analysis only with single chain files?
Thanks in advance for your help :)

Guillem Sanchez

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_phbradley_tcr-2Ddist_issues_41&d=DwMCaQ&c=eRAMFD45gAfqt84VtBcfhfEazhEXT91ASHynm_9f1N0&r=OoOdU4GyDM4g0P0UJHufcJpPOVmpY9zfZYFqEZ7QEzw&m=DeBGJQHMHRH-J6vpIH-OH2VPod3jy8zzMagKqbFW5nMWLFZfSRb5bSCo9Wi5-H8W&s=TgMkA3McRpwvtVFTTWDpACWe_J0e9wPSikSpZiwi4AE&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABBNCH43S3PH72OMF6S3OW3UZ67RLANCNFSM5NXMRLXQ&d=DwMCaQ&c=eRAMFD45gAfqt84VtBcfhfEazhEXT91ASHynm_9f1N0&r=OoOdU4GyDM4g0P0UJHufcJpPOVmpY9zfZYFqEZ7QEzw&m=DeBGJQHMHRH-J6vpIH-OH2VPod3jy8zzMagKqbFW5nMWLFZfSRb5bSCo9Wi5-H8W&s=TO9Tm6-mt0LwO0Kbx7YWZ5P2z-RyHmvdc3N4YC_XihI&e=>.
Triage notifications on the go with GitHub Mobile for iOS<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=eRAMFD45gAfqt84VtBcfhfEazhEXT91ASHynm_9f1N0&r=OoOdU4GyDM4g0P0UJHufcJpPOVmpY9zfZYFqEZ7QEzw&m=DeBGJQHMHRH-J6vpIH-OH2VPod3jy8zzMagKqbFW5nMWLFZfSRb5bSCo9Wi5-H8W&s=FJmQepC8b8KAt7-DOxoyZSvpFMjQ4WXG3cxAAnPyK64&e=> or Android<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=eRAMFD45gAfqt84VtBcfhfEazhEXT91ASHynm_9f1N0&r=OoOdU4GyDM4g0P0UJHufcJpPOVmpY9zfZYFqEZ7QEzw&m=DeBGJQHMHRH-J6vpIH-OH2VPod3jy8zzMagKqbFW5nMWLFZfSRb5bSCo9Wi5-H8W&s=-bBi6U0CLY4kHRCN2c6ypdjoHdeUIaS3r8NdV6lR-n0&e=>.

**Comment by guillemsanchezsanchez1996** [comment]
Dear Phil,
Thanks a lot for your fast answer! I have only TCR sequence
[AB1A.txt](https://github.com/phbradley/tcr-dist/files/8017484/AB1A.txt)
This is an example of a Mixcr processed file, as you can see same clones are pooled in the same rows.
One can also convert it to vdjtools file and simplify quite a lot the output, but the principle is the same (rows with accumulated counts of the distinct single chains).
[NFF.AB1A.txt](https://github.com/phbradley/tcr-dist/files/8017494/NFF.AB1A.txt)
Hope it helps,

Guillem

**Comment by guillemsanchezsanchez1996** [comment]
Hello again Phil,
Do you think is it feasable to try to transform these vdjtools/mixcr files? Or you think this single chain data cannot be used in your package to search cdr3 motifs?

Best,
Guillem

**Comment by phbradley** [comment]
Hi Guillem,
One more follow-up question: how many TCRs are you looking to analyze? Three other tools that might be more appropriate for single-chain data are


  *   ALICE (Walczak lab)
  *   TCRnet (from VDJtools folks)
  *   GLIPH2 (Davis Lab)

Take care,
Phil

________________________________

**Comment by guillemsanchezsanchez1996** [comment]
Dear Phil,
Thanks for the suggestions
I have 12 samples with around 100 to 500 TCR sequences. Do you think is it feasable? Is there a way to adapt the output lf mixcr to the input data for tcr-dist
Guillem

**Comment by agartland** [comment]
Hi Guillem
You may also want to take a look at tcrdist3, which is more flexible with its inputs and can do many of the same kinds of analyses.
All best,
Andrew

https://tcrdist3.readthedocs.io/en/latest/

https://elifesciences.org/articles/68605

**Comment by phbradley** [comment]
Yes! Oh, I just now realized that this was not a "conga" question but a "tcrdist" question. Sorry, Guillem, I have been getting many github emails about a different software project, and I was interpreting your email as pertaining to that one. For tcrdist-type analyses, tcrdist3 is absolutely the way to go, and as Andrew said it has some very flexible I/O.

I'm sorry for not realizing my mistake earlier.

Take care,
Phil

________________________________

**Comment by guillemsanchezsanchez1996** [comment]
Dear Andrew and Phil,
Thank you both for your suggestions. I was awared of tcrdist3 and I managed to run several functions with my data. I was interested in tcr-dist basically because of the nice tree representation of the cdr3 motifs that you do in your paper of 2017.

Best
Guillem
